planetscale/imports.go

```diff
@@ -0,0 +1,322 @@
+}
+
+func (d *dataImportsService) StartDataImport(ctx context.Context, request *StartDataImportRequest) (*DataImport, error) {
+	request.Source.SSLMode = request.Source.SSLVerificationMode.String()
+	path := fmt.Sprintf("/v1/organizations/%s/data-imports/new", request.Organization)
+	req, err := d.client.newRequest(http.MethodPost, path, request)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	resp := &DataImport{}
+	if err := d.client.do(ctx, req, &resp); err != nil {
+		return nil, err
+	}
+
+	return resp, nil
+}
+
+func (d *dataImportsService) GetDataImportStatus(ctx context.Context, getReq *GetImportStatusRequest) (*DataImport, error) {
+	path := fmt.Sprintf("/v1/organizations/%s/databases/%s", getReq.Organization, getReq.Database)
+	req, err := d.client.newRequest(http.MethodGet, path, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating request for get database")
+	}
+
+	db := &Database{}
+	err = d.client.do(ctx, req, &db)
+	if err != nil {
+		return nil, err
+	}
+
+	if db.DataImport.ID == "" {
+		return nil, errors.Errorf("Database %s is not importing data", getReq.Database)
+	}
+
+	db.DataImport.ParseState()
+	return &db.DataImport, nil
+}
+
+func (d *dataImportsService) CancelDataImport(ctx context.Context, cancelReq *CancelDataImportRequest) error {
+	path := fmt.Sprintf("%s/cancel", dataImportAPIPath(cancelReq.Organization, cancelReq.Database))
+	req, err := d.client.newRequest(http.MethodPost, path, nil)
+	if err != nil {
+		return errors.Wrap(err, "error creating http request")
+	}
+
+	if err := d.client.do(ctx, req, nil); err != nil {
+		return err
+	}
+
+	return nil
+}
+func (d *dataImportsService) MakePlanetScalePrimary(ctx context.Context, request *MakePlanetScalePrimaryRequest) (*DataImport, error) {
+	path := fmt.Sprintf("%s/begin-switch-traffic", dataImportAPIPath(request.Organization, request.Database))
+	req, err := d.client.newRequest(http.MethodPost, path, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	resp := &DataImport{}
+	if err := d.client.do(ctx, req, &resp); err != nil {
+		return nil, err
+	}
+	resp.ParseState()
+	return resp, nil
+}
+
+func (d *dataImportsService) MakePlanetScaleReplica(ctx context.Context, request *MakePlanetScaleReplicaRequest) (*DataImport, error) {
+	path := fmt.Sprintf("%s/begin-reverse-traffic", dataImportAPIPath(request.Organization, request.Database))
+	req, err := d.client.newRequest(http.MethodPost, path, nil)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating http request")
+	}
+
+	resp := &DataImport{}
+	if err := d.client.do(ctx, req, &resp); err != nil {
+		return nil, err
+	}
+	resp.ParseState()
+	return resp, nil
+}
+
+func (d *dataImportsService) DetachExternalDatabase(ctx context.Context, request *DetachExternalDatabaseRequest) (*DataImport, error) {
+	path := fmt.Sprintf("%s/cleanup", dataImportAPIPath(request.Organization, request.Database))
```

Comment:
Should we change this API endpoint to be something more semantic? Like, `detach-external-database` or `detach-external`? `cleanup` is ambiguous to the consumer, whereas we have internal knowledge of what it means.